apidef/oas/server.go

```diff
@@ -265,3 +290,132 @@ func (dt *DetailedTracing) Fill(api apidef.APIDefinition) {
 func (dt *DetailedTracing) ExtractTo(api *apidef.APIDefinition) {
 	api.DetailedTracing = dt.Enabled
 }
+
+// Event holds information about individual event to be configured on the API.
+type Event struct {
+	Enabled bool            `json:"enabled" bson:"enabled"`
+	Type    tykevent.Event  `json:"type" bson:"type"`
+	Action  tykevent.Action `json:"action" bson:"action"`
+	ID      string          `json:"id,omitempty" bson:"id,omitempty"`
+
+	WebhookCore
```

Comment:
I asume this should have better naming (`WebhookEvent`)? Given the log events, and custom events, maybe we should embed Event into more particular <T>Event{} structs?

Reply:
didn't get you. is the suggestion to make them fields over embedding? or just rename `WebhookCore` to `WebhookEvent`?